assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { RichText } from '@wordpress/block-editor';
+import { Button, Spinner } from '@wordpress/components';
+import { RawHTML, useEffect, useRef, useState } from '@wordpress/element';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { PostSelector } from '../../components';
+import useElementClickDetector from './element-click-detector';
+
+const AttachmentContent = ( props ) => {
+	const [ selectedPost, setSelectedPost ] = useState( null );
+	const [ failedToFetch, setFailedToFetch ] = useState( false );
+	const [ searchValue, setSearchValue ] = useState( '' );
+	const [ isFetching, setIsFetching ] = useState( false );
+	let fetchRequest, isStillMounted;
+
+	const {
+		attributes,
+		setAttributes,
+		toggleAttachment,
+	} = props;
+
+	const {
+		attachmentClass,
+		postId,
+		postType,
+		title,
+		wrapperStyle,
+	} = attributes;
+
+	const fetchSelectedPost = () => {
+		isStillMounted = true;
+		if ( postId ) {
+			setIsFetching( true );
+			const currentFetchRequest = fetchRequest = apiFetch( {
+				path: `/wp/v2/${ postType }s/${ postId }`,
+			} ).then(
+				( post ) => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( post );
+						setFailedToFetch( false );
+						setIsFetching( false );
+					}
+				}
+			).catch(
+				() => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( null );
+						setFailedToFetch( true );
+						setIsFetching( false );
+					}
+				}
+			);
+		}
+	};
+
+	useEffect( () => {
+		return () => {
+			isStillMounted = false;
+		};
+	}, [] );
+
+	useEffect( () => {
+		fetchSelectedPost();
+	}, [ postId ] );
+
+	const removePost = () => {
+		setFailedToFetch( false );
+		setAttributes( { postId: null } );
+		setSelectedPost( null );
+	};
+
+	// Since the Page Attachment content is "out of it's borders" then clicking
+	// on the links might sometimes lose focus instead. Using refs for workaround.
+	const removeLinkRef = useRef( null );
+	const closeButtonRef = useRef( null );
+
+	useElementClickDetector( closeButtonRef, () => {
+		toggleAttachment( false );
+	} );
+
+	useElementClickDetector( removeLinkRef, null, true );
+
+	return (
+		<div className="attachment-container">
+			<div className="attachment-wrapper">
+				<div className="attachment-header">
+					<span
```

Comment:
Why not use a iconbutton here?

Reply:
This is the same close button that is used on the frontend by AMP, to make sure they look the same.

Reply:
Might worth putting a comment on the code to state that.